EIPS/eip-1613.md

```diff
@@ -67,15 +67,19 @@ Implementing a `RelayRecipient` contract:
 * Maintain a small balance of ETH gas prepayment deposit in `RelayHub`. Can be paid directly by the `RelayRecipient` contract, or by the dapp's owner on behalf of the `RelayRecipient` address. 
   The dapp owner is responsible for ensuring sufficient balance for the next transactions, and can stop depositing if something goes wrong, thus limiting the potential for abuse of system bugs. In DAO usecases it will be up to the DAO logic to maintain a sufficient deposit.
 * Use `get_sender()` and `get_data()` instead of `msg.sender` and `msg.data`, everywhere. `RelayRecipient` provides these functions and gets the information from `RelayHub`.
-* Implement a `may_relay(address relay, address from, uint32 gasPrice, bytes transaction)` view function that returns `true` if and only if it is willing to accept a transaction and pay for it. 
-  `may_relay` is called by `RelayHub` as a view function when a `Relay` inquires it, and also during the actual transaction. Transactions are reverted if `false`, and `Relay` only gets compensated for transactions (whether successful or reverted) if `may_relay` returns `true`. Some examples of `may_relay()` implementations:
+* Implement a `accept_relayed_call(address relay, address from, uint32 gasPrice, bytes encoded_function)` view function that returns **zero** if and only if it is willing to accept a transaction and pay for it. 
+  `accept_relayed_call` is called by `RelayHub` as a view function when a `Relay` inquires it, and also during the actual transaction. Transactions are reverted if **non-zero**, and `Relay` only gets compensated for transactions (whether successful or reverted) if `accept_relayed_call` returns **zero**. Some examples of `accept_relayed_call()` implementations:
     * Whitelist of trusted dapp members.
     * Balance sheet of registered users, maintained by the dapp owner. Users pay the dapp with a credit card or other non-ETH means, and are credited in the `RelayRecipient` balance sheet. 
       Users can never cost the dapp more than they were credited for.
     * Whitelist of known transactions used for onboarding new users. This allows certain anonymous calls and is subject to Sybil attacks. 
       Therefore it should be combined with a restricted gasPrice, and a whitelist of trusted relays, to reduce the incentive for relays to create bogus transactions and rob the dapp's prepaid gas deposit. 
       Dapps allowing anonymous onboarding transactions might benefit from registering their own `Relay` and accepting anonymous transactions only from that `Relay`, whereas other transactions can be accepted from any relay. 
       Alternatively, dapps may use the balance sheet method for onboarding as well, by applying the methods suggested in the attacks/mitigations section below.
+* Implement `post_relayed_call(address relay, address from, uint32 gasPrice, bytes encoded_function, uint gas_used)`
```

Comment:
Ditto. Doesn't seem to match the function signature in the implementation PR: (relay, from, encoded_function, success, used_gas).

gasPrice appears here but not in the implementation. It may actually be redundant here, since it's not a view function (unlike accept_relayed_call), so gasPrice can be taken from tx. Consider removing gasPrice here.

and transaction_fee is missing, both in the doc and in the implementation. If post_relayed_call is used to actually charge the sender somehow, then there are three factors to consider: tx.gasprice, gas_used, and relay's transaction_fee.